compiler/src/main/java/com/squareup/anvil/compiler/codegen/CodeGenerationExtension.kt

```diff
@@ -90,7 +94,21 @@ internal class CodeGenerationExtension(
 
     fun Collection<CodeGenerator>.generateCode(files: Collection<KtFile>): Collection<KtFile> =
       flatMap { codeGenerator ->
-        codeGenerator.generateCode(codeGenDir, anvilModule, files).toKtFile()
+        codeGenerator.generateCode(codeGenDir, anvilModule, files)
+          .onEach { generatedFile ->
+            val path = generatedFile.file.relativeTo(codeGenDir).path
+            if (!generatedFilePaths.add(path)) {
+              throw AnvilCompilationException(
+                "There were duplicate generated files. Generating and overriding the same file " +
```

Comment:
Should this be 'overwriting' instead of 'overriding'?

Reply:
Done